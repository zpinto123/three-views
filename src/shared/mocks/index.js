export const books = [
  { bookId: 1, author: "john", title: "john book", price: 5 },
  { bookId: 2, author: "joana", title: "joana book", price: 7 }
];
