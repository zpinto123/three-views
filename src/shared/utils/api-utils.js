import axios from "axios";
import api from "../../services/config";
import { STATUS_OK } from "../constants/api-status-codes";

const post = (query, variables) =>
  axios({
    url: `${api.base}?`,
    method: "post",
    data: { query, variables }
  })
    .then(verifyResponse)
    .catch(handleErrors);

const verifyResponse = res => {
  if (!res) return null;
  if (res.status !== STATUS_OK || !res.data) {
    return res.status;
  }
  return res.data.data;
};

const handleErrors = error => error;

export { post };
