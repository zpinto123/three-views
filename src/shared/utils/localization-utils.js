import i18next from 'i18next';
import { withTranslation } from 'react-i18next';

const setupResources = (namespace, resources) => {
  for (let lang of Object.keys(resources)) {
    if (!i18next || !i18next.hasResourceBundle) return;
    if (!i18next.hasResourceBundle(lang, namespace))
      i18next.addResourceBundle(lang, namespace, resources[lang]);
  }
};

const withLocalization = (
  Component,
  resources,
  name,
  withoutNamespace = false
) => {
  setupResources(name, resources);
  if (withoutNamespace) return withTranslation()(Component);
  return withTranslation(name)(Component);
};

export { withLocalization, setupResources };
