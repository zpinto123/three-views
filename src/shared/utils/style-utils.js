import styled from 'styled-components';

const style = (Component, styles = '', attrs = {}) =>
  styles
    ? styled(Component).attrs(attrs)`
        ${styles}
      `
    : Component;

export { style };
