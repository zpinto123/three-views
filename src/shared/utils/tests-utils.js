import configureStore from "redux-mock-store";
import books from "../state/books/initial-state";

const mockStore = configureStore();
const store = mockStore({ books });

export { store };
