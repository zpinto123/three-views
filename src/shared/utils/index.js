import * as apiUtils from './api-utils';
import * as localizationUtils from './localization-utils';
import * as styleUtils from './style-utils';

export { apiUtils, localizationUtils, styleUtils };
