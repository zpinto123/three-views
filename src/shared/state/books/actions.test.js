import { store } from "../../utils/tests-utils";
import { books } from "../../mocks";

const setBooks = books => ({
  type: "SET_BOOKS",
  payload: { books }
});

describe("action creators", () => {
  it("Saves books to redux if getBooks is successful", () => {
    store.dispatch(setBooks(books));
    expect(store.getActions()).toMatchSnapshot();
  });
});
