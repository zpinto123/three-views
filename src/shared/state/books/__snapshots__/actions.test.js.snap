// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action creators Saves books to redux if getBooks is successful 1`] = `
Array [
  Object {
    "payload": Object {
      "books": Array [
        Object {
          "author": "john",
          "bookId": 1,
          "price": 5,
          "title": "john book",
        },
        Object {
          "author": "joana",
          "bookId": 2,
          "price": 7,
          "title": "joana book",
        },
      ],
    },
    "type": "SET_BOOKS",
  },
]
`;
